src/mono/mono/mini/aot-runtime.c

```diff
@@ -2275,7 +2275,7 @@ load_aot_module (MonoAssemblyLoadContext *alc, MonoAssembly *assembly, gpointer
 
 		addr = amodule->mem_begin;
 		g_assert (addr);
-		len = amodule->mem_end - amodule->mem_begin;
+		len = (int)(amodule->mem_end - amodule->mem_begin);
```

Comment:
this one is a bit weird to me in general, I'm not sure why we need `addr` and `len` since for calculating `page_end` we're adding them back together so couldn't we just use `amodule->mem_end` instead?

Reply:
removed the intermediate variables